docs/tutorial.md

```diff
@@ -176,3 +177,86 @@ spec:
           serviceName: frontend
           servicePort: 80
 ```
+
+
+## Expose a WebSocket server
+
+As outlined in the Application Gateway v2 documentation - it [provides native support for the WebSocket and HTTP/2 protocols](https://docs.microsoft.com/en-us/azure/application-gateway/overview#websocket-and-http2-traffic). For both App Gateway and the Kubernetes Ingress - there is no user-configurable setting to selectively enable or disable WebSocket support.
+
+The Kubernetes deployment YAML below shows the minimum configuration used to deploy a WebSocket server, which is the same as deploying a regular web server:
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: websocket-server
+spec:
+  selector:
+    matchLabels:
+      app: ws-app
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        app: ws-app
+    spec:
+      containers:
+        - name: websocket-app
+          imagePullPolicy: Always
+          image: your-container-repo.azurecr.io/websockets-app
+          ports:
+            - containerPort: 8888
+      imagePullSecrets:
+        - name: azure-container-registry-credentials
+
+---
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: websocket-app-service
+spec:
+  selector:
+    app: ws-app
+  ports:
+  - protocol: TCP
+    port: 80
+    targetPort: 8888
+
+---
+
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: websocket-repeater
+  annotations:
+    kubernetes.io/ingress.class: azure/application-gateway
+spec:
+  rules:
+    - host: ws.contoso.com
+      http:
+        paths:
+          - backend:
+              serviceName: websocket-app-service
+              servicePort: 80
+```
+
+Given that all the prerequisites are fulfilled, and you have an App Gateway controlled by a K8s Ingress in your AKS, the deployment above would result in a WebSockets server exposed on port 80 of your App Gateway's public IP and the `ws.contoso.com` domain.
```

Comment:
Should we describe that the above webserver is configured to run websockets on `/ws` while `/` or `/health` runs an https server returning `200 OK` ?

Reply:
👍 . This is pretty important once we get the custom health probes documentation in.

Reply:
Looks like @akshaysngupta point above is covered in the section below?